_posts/2023-12-06-where-is-Strimzi-1.0.0.md

```diff
@@ -0,0 +1,86 @@
+We knew that it would take a long time and mean many different changes.
+Changes to Apache Kafka, its Admin API and the way it is managed and configured.
+But also changes to Strimzi.
+For example, without ZooKeeper, the layout of our `Kafka` custom resource with its dedicated `.spec.kafka` and `.spec.zookeeper` sections does not make sense anymore.
+So we decided to not release Strimzi 1.0.0 and wait for the ZooKeeper removal to be finished.
+
+It turned out that the ZooKeeper removal from Apache Kafka took longer than we initially expected.
+In fact - it is still in progress - although it seems like it is slowly coming to an end and hopefully, next year will see the release of Apache Kafka 4.0 without ZooKeeper support.
+At the time of writing this blog post, Strimzi has 0.38.0 as its last release.
+That means 38 minor releases without releasing 1.0.0.
+
+So, looking back, the decision we took a long time ago was probably a mistake.
+If I had the chance to go back and change it, I would surely propose to do the 1.0.0 release already back then and now we would be talking only about _when to do the 2.0.0 release_.
+
+### Does it mean Strimzi is not production-ready?
+
+Absolutely not!
+We realize the 0.x version might seem scary and might put people off.
+However, the version number itself does not say much about the project maturity or about how good the project is.
+All of us in Strimzi believe that Strimzi production-ready.
+And you can check the logos of some of our production users on our website as proof.
+
+And despite being only at 0.x version, we do our best to maintain backward compatibility on the API level as well in how the Strimzi operators work.
+So you should not have anything to worry about!
+
+### How do we get out of it?
+
+So what is the plan for Strimzi 1.0.0?
+When we realized that the ZooKeeper removal would take much longer than originally anticipated, we had two options.
+Change our mind and release 1.0.0.
+Or stick with our decision.
+And since the KRaft implementation was progressing in the Apache Kafka community and it seemed to be closer and closer, we decided to stick to our decision.
+Maybe that was another mistake, who knows?
+Things always seem like they are almost there.
+
+But hopefully, the _next year_ will really be the **next year** when it happens.
+If everything goes according to the _latest_ plans, Apache Kafka 4.0 without ZooKeeper support will be released in 2024.
+Once it is released and supported by Strimzi, we will proceed and update the `Kafka` custom resource to reflect this change.
+The updated `Kafka` resource will be used for the Strimzi `v1` API.
+And the `v1` API will be hopefully soon followed by the Strimzi 1.0.0 release.
+This might take another few months to stabilize the APIs, catch the last bugs and so on.
+So depending on when the Kafka 4.0.0 release happens, it might not be 2024 anymore.
+But then we will be finally there - at version 1.0.0.
```

Comment:
I think we should say that there will inevitably be some breaking changes at that point, but we will document them and provide migration tools. Do you think we will be able to automate the migration from the operator?